lib/kazoo/partition.rb

```diff
@@ -16,7 +16,7 @@ def replication_factor
     end
 
     def preferred_leader
-      @replicas.first
+      @replicas.compact.first
```

Comment:
Sorry, I was AFK for a week.
`replicas` is passed into the constructor for partition in topic.
Around line 171 it gets the list of replica ids from each partition of the topic, and indexes into `cluster.brokers[id]` to try and turn the `id` into a broker.

If the cluster doesn't contain that broker object, then that returns `nil`.

Should it be filtered out at that level, and only _up_ brokers be in the replicas array, or should the cluster know about the down broker, even if it's down?